R/MsBackendTimsTof-functions.R

```diff
@@ -77,9 +77,9 @@
   msg <- .valid_required_columns(x@indices, c("frame", "file"))
   if ("file" %in% colnames(x@indices) &&
       any(!x@indices[, "file"] %in% seq_len(length(x@fileNames))))
```

Comment:
If the indices not in `seq_len` they have to be larger than `length`. Maybe `any(x@indices[, "file"] > length(x@fileNames))` is simpler to understand?

Reply:
I now notice that there is a problem arising when we subset the object (also with the current version). When we subset the object we also subset the `@fileNames` slot. If e.g.  `@fileNames = c("a", "b", "c")` and subsetting removes all spectra with `@indices[, "file"] == 2` then `@fileNames = c("a", "c")` and that causes that condition not to work properly. Should we avoid to subset the `@fileNames` slot? Or update `@indices[, "file"]` when a situation like that presents during subsetting? Or maybe using a named numeric for `@fileNames` establishing the connection between file index and file name would help?

Reply:
Ah, yes. Good point. So, for the subsetting to work properly you need to subset `@indices`, the `@fileNames` and then update also the `@indices[, "file"]` (using `match`? or maybe you find a more efficient way). Indices in there should always just be from 1 to `length(@fileNames)`.

Reply:
hmmm. the named numeric option. that would maybe be more efficient than replacing the `@indices[, "file"]`, which I would expect to take *very* long in this rather large files. yes, maybe try that. But make sure this also works if you have duplicated file names and all other use cases.

Reply:
Ok and yes it should work with duplicated file names too. Alternatively we could also just avoid to modify the `@fileNames` during subsetting and when needed retrieve on the fly the indices of those that are ok with unique(object@indices[, “file”]) but maybe it could create confusion to have also file names in `@fileNames` for which there’s no more spectra in the object?